web/src/compose_validate.ts

```diff
@@ -38,11 +38,13 @@ import * as util from "./util.ts";
 
 let user_acknowledged_stream_wildcard = false;
 let upload_in_progress = false;
-let no_channel_selected = false;
-let missing_topic = false;
-let no_private_recipient = true;
 let no_message_content = false;
 let message_too_long = false;
+// Since same functions are used for both compose and message edit,
+//  we need to track when we are validating compose box.
+let is_validating_compose_box = false;
+let disabled_send_tooltip_message = "";
+let posting_policy_error_message = "";
```

Comment:
This global data structure model is kinda fragile. We will probably want to move it to these properties being all fields of an object type that we can one for compose and one for each edit widget.

Reply:
I think ideally we would add a new type that encapsulates these and the parameters that get passed around everywhere. But that can be a follow-up project, since this PR does fix an important bug.

Reply:
Okay, added that to my TODO.